instrumentation/rails/lib/opentelemetry/instrumentation/rails/span_subscriber.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Rails
+      # The SpanSubscriber is a special ActiveSupport::Notification subscription
+      # handler which turns notifications into generic spans, taking care to handle
+      # context appropriately.
+      class SpanSubscriber
+        ALWAYS_VALID_PAYLOAD_TYPES = [TrueClass, FalseClass, String, Numeric, Symbol].freeze
+
+        def initialize(name:, tracer:)
+          @span_name = name.split('.')[0..1].reverse.join(' ').freeze
+          @tracer = tracer
+        end
+
+        def start(_name, _id, payload)
+          span = @tracer.start_span(@span_name, kind: :internal)
```

Comment:
Is there a situation where we'd use a span subscriber to produce DB client spans (for example)? We've done that previously at Shopify, but this project has lower level instrumentation for a bunch of DBs, so we're not doing that anymore.

Reply:
I think you certainly could. I don't know if we _would_ or not but I don't see anything prohibiting it.

Reply:
Should we allow passing in the span type as a config option? I'm also totally OK with punting that to a subsequent PR.

Reply:
I think we should punt. Not that it would be terribly difficult to add or anything, but we also don't have a real need for it yet (unless you know of something coming up that I don't).